lib/ansible/modules/network/eos/eos_l2_interfaces.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+##############################################
+#                 WARNING                    #
+##############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+##############################################
+
+"""
+The module file for eos_l2_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'network'
+}
+
+DOCUMENTATION = """
+---
+module: eos_l2_interfaces
+version_added: 2.9
+short_description: Manages Layer-2 interface attributes of Arista EOS devices
+description: This module provides declarative management of Layer-2 interface on Arista EOS devices.
+author: Nathaniel Case (@qalthos)
+options:
+  config:
+    description: A dictionary of Layer-2 interface options
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+        - Full name of interface, e.g. Ethernet1.
+        type: str
+        required: True
+      access:
+        description:
+        - Switchport mode access command to configure the interface as a layer 2 access.
+        type: dict
+        suboptions:
+          vlan:
+            description:
+            - Configure given VLAN in access port. It's used as the access VLAN ID.
+            type: int
+      trunk:
+        description:
+        - Switchport mode trunk command to configure the interface as a Layer 2 trunk.
+        type: dict
+        suboptions:
+          native_vlan:
+            description:
+            - Native VLAN to be configured in trunk port. It is used as the trunk native VLAN ID.
+            type: int
+          trunk_allowed_vlans:
+            description:
+            - List of allowed VLANs in a given trunk port. These are the only VLANs that will be
+              configured on the trunk.
+            type: list
+  state:
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    default: merged
+    description:
+    - The state the configuration should be left in
+    type: str
```

Comment:
Please add a notes section mentioning the eos version tested, connection type used and link to eos platform guide.
Refer https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L103